src/System.IO.FileSystem/tests/Directory/EnumerableTests.cs

```diff
@@ -31,6 +31,38 @@ public void FileEnumeratorIsThreadSafe()
             {
             }
         }
+        
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public void EnumerateDirectories_NonBreakingSpace()
+        {
+            string TestDirectory = GetTestFilePath();
+            DirectoryInfo di = new DirectoryInfo(TestDirectory);
+            di.Create();
+            di.CreateSubdirectory("\u00A0");
+
+            var diQueue = new Queue<DirectoryInfo>();
+            diQueue.Enqueue(di);
+            int count = 0;
+
+            while (diQueue.Count != 0)
```

Comment:
This seems unnecessarily complicated and is missing some important validation. The fundamental problem was that the trailing whitespace char was eaten in `Path.GetFullPath()`.  `TrimmedPaths` has related tests. The fundamental test here is that you get back the subdirectory `"\u00A0"` and can use it successfully. You aren't checking the path name you got back. You need to create three subdirs to validate this correctly and not create problems for other tests.  Create something like `"\\A\\\u00A0\\B"` under the `TestDirectory` then enumerate A recursively and see that you get the two subdir names back.

Reply:
Don't use A for the first name, however. Use 'GetTestFileName()` to ensure that you don't collide with other test methods.